### DISCUSSION TITLE
Discussion

### DISCUSSION THREAD

**Discussion by leibaoer** [open-edge-platform/anomalib#585] [Q&A] [answered]
I am a student comes from China,so my English is not very good.I want to ask you another question, and look forward to your reply.My defects can be displayed on the heatmaps, but they can't be circled on the pre mask. What parameters do I need to adjust and turn them up or down? My test results are not very good,like these.
![image](https://user-images.githubusercontent.com/113488512/192074639-7b22d77e-5ba4-4f2b-9bfa-b097c3775fa8.png)
![image](https://user-images.githubusercontent.com/113488512/192074642-75cddbc4-0a1c-4faa-8f80-533c01d92287.png)
![image](https://user-images.githubusercontent.com/113488512/192074643-18546991-13ba-43eb-81d9-b5d218ea97c9.png)
![image](https://user-images.githubusercontent.com/113488512/192074648-14e49c0f-6bd2-47da-8470-1dc70b3710a7.png)

**Comment by ashwinvaidya17** [comment, marked as answer]
Copying the answer from https://github.com/openvinotoolkit/anomalib/discussions/579#discussioncomment-3731491

The segmentation maps are generated from the threshold. So while the heatmap might show something the threshold determines the value above which the predicted values are considered as anomalies. Normalization ensures that this threshold is centred to 0.5. So, when generating the segmentation masks the anomalous regions are the ones with values greater than 0.5.
1. Since this threshold is calculated from the validation images, I would suggest adding more images with ground truth masks in your dataset.
2. You can also have a look at the adaptive threshold class https://github.com/openvinotoolkit/anomalib/blob/main/anomalib/utils/metrics/adaptive_threshold.py#L10. Currently, the threshold is computed based on optimal f1 score. Maybe you use a different metric here.

**Reply by djdameln** [reply]
3. There may be several reasons why the adaptive threshold does not produce satisfactory results on a qualitative level. In this case you may want to experiment with manually finetuning the anomaly score threshold. To use a manual threshold, the `metrics.threshold.method` should be set to `manual` after which the threshold values for image- and pixel-level detections can be configured using the `manual_image` and `manual_pixel` parameters. Please note that the threshold is applied to the model outputs before normalization, so the range of appropriate threshold values is unbounded and may differ differ between algorithms.
